src/crewai/agents/agent_builder/base_agent.py

```diff
@@ -127,11 +127,46 @@ class BaseAgent(ABC, BaseModel):
         default=None, description="Maximum number of tokens for the agent's execution."
     )
 
+    @model_validator(mode="before")
+    @classmethod
+    def process_config(cls, values):
+        config = values.get("config", {})
+        if not config:
+            return values
+
+        for key, value in config.items():
+            if key in cls.model_fields:
```

Comment:
How do you feel about early returns? In this case, early continue? I like it to prevent block nesting.

Reply:
Refactored for early breaks with continues!